src/hinting.ts

```diff
@@ -195,7 +195,10 @@ function buildHintsSimple(els: Element[], onSelect: HintSelectedCallback) {
 
 function buildHintsVimperator(els: Element[], onSelect: HintSelectedCallback) {
     let names = hintnames(els.length)
-    const filterableTextFilter = new RegExp('[' + config.get('hintchars') + ']|[^[:alnum:]]', 'gi')
+    // escape the hintchars string so that strange things don't happen
+    // when special characters are used as hintchars (for example, ']')
+    const escapedHintChars = config.get('hintchars').replace(/^\^|[-\\\]]/g, "\\$&")
+    const filterableTextFilter = new RegExp('[' + escapedHintChars + ']', 'gi')
     for (let [el, name] of izip(els, names)) {
         let ft = elementFilterableText(el)
         // strip out non-alphanumeric characters and hintchars.
```

Comment:
Why not omit all this regex stuff?

`Array.from(ft).filter(char => hintchars.indexOf(char) == -1).join('')`

Reply:
My snippet returns `ft` with every character in hintchars removed, which I think is what the regex does, but I haven't read the regex closely to check.

Reply:
Yes, the intent is to remove from the filterable text all characters that are in the hintchars set.

I didn't look too hard at filter-join pipelines for this because I was worried about performance. I haven't done a real survey, but I know that I've run into a couple sites with elements with tens of k of inner html - it can happen when someone puts a target on an element that has a ton of stuff inside it. I suppose that we could just wait for profiling (or reports of slowness) instead of optimizing prematurely, or just truncate, or something else, though?

Reply:
I agree that I'm not happy with the regex silliness. :/

Reply:
> Programmers waste enormous amounts of time thinking about, or worrying about, the speed of noncritical parts of their programs, and these attempts at efficiency actually have a strong negative impact when debugging and maintenance are considered. We should forget about small efficiencies, say about 97% of the time: **premature optimization is the root of all evil**. Yet we should not pass up our opportunities in that critical 3%.

\-- *Donald Knuth*

We should go with the simpler option and consider higher level optimisations such as truncation or avoiding precompution entirely, if optimisation is required.

Out of curiosity, I decided to benchmark and RegExp is a bit faster, but they're both pretty fast: https://jsperf.com/filter-chars-from-string

On my system, filtering a .7MB string takes ~62ms via the array method and ~50ms with RegExp. For reference, human reaction time is about 200ms.

Reply:
Human reaction time isn't a particularly good benchmark for UI - I'd aim for screen refresh (~16ms) for guaranteed imperceptibility.

Reply:
Sure, but I don't care about getting it down that low here. Doubtful we could get the total down that far.

Looking at the code some more, I would have had vimperator mode accept any character and move the filter function to the Hint, avoiding all precomputing. I think what you've done is fitting the new implementation a bit too strongly to the old model.

We could just slap a truncation at 1K characters (in case of big child nodes (data URIs, script tags, etc) and use this if you don't fancy refactoring.